order/views.py

```diff
@@ -1 +1,280 @@
-from django.shortcuts import render
+import json
+from datetime               import datetime
+
+from django.http            import JsonResponse
+from django.views           import View
+from django.core.exceptions import ObjectDoesNotExist
+from django.core.exceptions import MultipleObjectsReturned
+
+from user.utils             import login_decorator
+from user.models            import Address
+from product.models         import ProductStock 
+from .models                import Order, OrderProductStock, OrderStatus
+
+class CartView(View):
+    @login_decorator
+    def get(self, request):
+        try:
+            user = request.user
+            
+            if not Order.objects.filter(user=user, order_status=OrderStatus.objects.get(name='주문 전')).exists():
+                return JsonResponse({"message": "EMPTY"}, status=200)
+            
+            order          = Order.objects.get(user=user, order_status=OrderStatus.objects.get(name='주문 전'))
+            order_products = order.orderproductstock_set.filter(order=order)
+        
+            cart_product_list = [{
+                "category"        : order_product.product_stock.product.category.menu.name,
+                "productId"       : order_product.product_stock.product.id,
+                "productName"     : order_product.product_stock.product.name,
+                "productSubName"  : order_product.product_stock.product.sub_name,
+                "productStockId"  : order_product.product_stock.id,
+                "productSize"     : order_product.product_stock.size,
+                "productPrice"    : order_product.product_stock.price,
+                "productStock"    : order_product.product_stock.stock,
+                "productImageUrl" : order_product.product_stock.product.image_set.get(is_main=True).image_url,
+                "productQuantity" : order_product.quantity
+                } for order_product in order_products
+            ]
```

Comment:
```suggestion
            cart_product_list = [{
                "category"        : order_product.product_stock.product.category.menu.name,
                "productId"       : order_product.product_stock.product.id,
                "productName"     : order_product.product_stock.product.name,
                "productSubName"  : order_product.product_stock.product.sub_name,
                "productStockId"  : order_product.product_stock.id,
                "productSize"     : order_product.product_stock.size,
                "productPrice"    : order_product.product_stock.price,
                "productStock"    : order_product.product_stock.stock,
                "productImageUrl" : order_product.product_stock.product.image_set.get(is_main=True).image_url,
                "productQuantity" : order_product.quantity
                } for order_product in order.orderproductstock_set.filter(order=order)]
```
list comprehension을 사용해서 잘 바꾸셨습니다.~~!!!!

order_products라는 변수가 한 번만 사용되고 있습니다. 이럴 경우 굳이 변수에 담지 않고 사용하는 것을 권장 합니다.

Reply:
네 수정하였습니다~